tests/sync_filter_test.go

```diff
@@ -2,35 +2,47 @@ package tests
 
 import (
 	"encoding/json"
+	"io/ioutil"
 	"testing"
 
+	"github.com/tidwall/gjson"
+
 	"github.com/matrix-org/complement/internal/b"
-	"github.com/matrix-org/complement/internal/match"
-	"github.com/matrix-org/complement/internal/must"
+	"github.com/matrix-org/complement/internal/client"
 )
 
 func TestSyncFilter(t *testing.T) {
 	deployment := Deploy(t, b.BlueprintAlice)
 	defer deployment.Destroy(t)
 	authedClient := deployment.Client(t, "hs1", "@alice:hs1")
 	// sytest: Can create filter
-	reqBody, err := json.Marshal(map[string]interface{}{
-		"room": map[string]interface{}{
-			"timeline": map[string]int{
-				"limit": 10,
+	t.Run("Can create filter", func(t *testing.T) {
+		reqBody, err := json.Marshal(map[string]interface{}{
+			"room": map[string]interface{}{
+				"timeline": map[string]int{
+					"limit": 10,
+				},
 			},
-		},
+		})
+		if err != nil {
+			t.Fatalf("failed to marshal JSON request body: %s", err)
+		}
+		create_filter(authedClient, reqBody, t, "@alice:hs1")
 	})
+}
+
+func create_filter(authedClient *client.CSAPI, reqBody []byte, t *testing.T, userID string) string {
```

Comment:
please don't use `_` in function names. I don't see why this function needs to be separate, where else do we call it?